Cauldron/Assets/Scripts/ExperimentController.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CauldronCodebase;
+using UnityEngine;
+using UnityEngine.UI;
+using Random = UnityEngine.Random;
+
+public class ExperimentController : MonoBehaviour
+{
+    [SerializeField] private RecipeBook recipeBook;
+    [SerializeField] private AnimalsFilter animalsFilter;
+    [SerializeField] private RootsFilter rootsFilter;
+    [SerializeField] private MushroomsFilter mushroomsFilter;
```

Comment:
I would make the master filter script with event FilterChanged and Ingredients[] GetFilter() to separate it from the controller.